### PR TITLE
uftrace: Add --no-args option to hide args/retval

### PR DESCRIPTION
**Pull request by honggyukim**
It needs to provide a way to hide args/retval to reduce the amount of
dump output.  It has to be implemented for replay command later on.

Signed-off-by: Honggyu Kim <honggyu.kp@gmail.com>

### CODE REVIEW

**Comment by honggyukim** [comment]
It's a draft implementation and it has to be implemented for replay command as well.

**Comment by namhyung** [comment]
> It's a draft implementation and it has to be implemented for replay command as well.

Well, you could simply pass the `opts->no_args` to `get_argspec_string()` and make it return an empty string.

**Comment by honggyukim** [comment]
> Well, you could simply pass the opts->no_args to get_argspec_string() and make it return an empty string.

Right. I just updated it that handles both replay and dump output. Thanks!

**Comment by honggyukim** [comment]
Okay. I will make it as `print_args`.

**Comment by honggyukim** [comment]
There is an internal macro `print_args` in `cmds/replay.c`, so I just updated it using `show_args` as you suggested. Thanks!

**Comment by namhyung** [comment]
Maybe we could make it global variable.  But some code is shared between uftrace binary and libmcount so it might be confusing if some setting has different value for some reason.

**Comment by namhyung** [comment]
And I'm hesitating to add new option everytime we introduce new features..

**Comment by honggyukim** [comment]
As I wrote in #599, it can be combined into `--args="off"` option.

**Comment by honggyukim** [comment]
If the option name `--no-args` is not sure yet, then would it be better to disable argument info in `dump --chrome` output as of now? If some arguments contains not printable characters, then chrome browser fails to load the `.json` file.  

It's already handled in 58ef456a, but it's not clearly fixed for some reasons.

**Comment by honggyukim** [comment]
@namhyung I just rebased and updated this PR again. Please have a look. Thanks!

File: uftrace.c [hunk 1]
```diff
@@ -130,6 +131,7 @@ __used static const char uftrace_help[] =
 "      --avg-self             Show average/min/max of self function time\n"
 "      --avg-total            Show average/min/max of total function time\n"
 "  -a, --auto-args            Show arguments and return value of known functions\n"
+"      --no-args              Do not show arguments and return value\n"
```
Comment (namhyung) [inline uftrace.c:134]: I know why you placed it here, but I think we should maintain it in alphabetical order.

**Review by namhyung — COMMENTED** [review]
Looks ok, one minor nit..

File: uftrace.c (hunk 1 above)
Comment (honggyukim) [reply uftrace.c]: Right. I just reordered it. Thanks.

**Review by namhyung — APPROVED** [review]
LGTM

**Pull request merged**